src/utils/offers/offerAggregateOptions.js

```diff
@@ -37,6 +40,22 @@ const offerAggregateOptions = (query, params, user) => {
     match['$or'] = [{ title: getRegex(search) }, ...additionalFields]
   }
 
+  if (authorId !== userId) {
+    if (authorRole) {
+      match[`author.status.${authorRole}`] = STATUS_ENUM[0]
+    } else {
+      match['$expr'] = {
+        $switch: {
+          branches: [
+            { case: { $eq: ['$authorRole', 'student'] }, then: { $eq: ['$author.status.student', 'active'] } },
+            { case: { $eq: ['$authorRole', 'tutor'] }, then: { $eq: ['$author.status.tutor', 'active'] } },
+            { case: { $eq: ['$authorRole', 'admin'] }, then: { $eq: ['$author.status.admin', 'active'] } }
```

Comment:
This code seems to be repeated. I suggest using loop here like following:

```js
Object.keys(roles).map(role => ({
     case: { $eq: ['$authorRole', role }, then: { $eq: [`$author.status.${role}`, 'active'] }
}))
```

You can find `roles` constant in const folder in `src`. Don't forget to omit `SUPERADMIN` role there

Reply:
Thanks for noticing, refactored